### PR TITLE
fix: clamp Add Cards For Today to the actual unseen card count

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Summary

- The "Add Cards For Today" dialog let users grant an arbitrarily large boost to today's new-card quota — even with 0 unseen cards available — and that number would be saved and rendered as the day's quota, even though it could never actually be fulfilled.
- Fixed with defense in depth across three layers:
  - **Write-side clamp**: `DayCountersStore.addExtraNewToday` now takes the live unseen count and clamps the boost so `newPerDay + extraNewToday - newShown` can never exceed it (also fixes repeated adds accumulating past capacity).
  - **Display cap**: `DojoViewModel`'s "N new" counter is now capped at the real unseen total via a new reactive `VocabularyDao.observeNewTotalCount()`, so an already-persisted oversized value can't be displayed either.
  - **UI guard**: `NumberInputDialog` gained a `maxValue` bound; the Add Cards For Today dialog now passes the actual remaining capacity, disabling OK entirely when nothing can be added.
- Also covered a related edge case: `New Cards Per Day` itself set higher than the number of unseen cards.

## Test plan

- Added/updated unit tests: `DayCountersStoreTest`, `SettingsViewModelTest`, `DojoViewModelTest`, `AddCardsForTodayIntegrationTest`, `NumberInputDialogTest`, `SettingsContentTest`.
- `./gradlew testDebugUnitTest` — full suite passes.
- `./gradlew ktlintCheck detekt` — clean.

---
_Generated by [Claude Code](https://claude.ai/code/session_016BEzWAoMWxL3oJN5fAchVu)_

### CODE REVIEW

File: app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt [hunk 1]
```diff
@@ -84,10 +84,20 @@ class DayCountersStore
             }
         }
 
-        suspend fun addExtraNewToday(amount: Int) {
+        // Clamps the boost so newPerDay + extraNewToday - newShown can never exceed
+        // [availableNew], the actual number of unseen cards left in the deck.
```
Comment (Vladyslav-Soldatenko) [inline app/src/main/java/com/procrastilearn/app/data/local/prefs/DayCountersStore.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/screens/settings/SettingsContentTest.kt [hunk 2]
```diff
@@ -188,6 +188,60 @@ class SettingsContentTest {
...
+
+        composeTestRule.onNodeWithText(string(R.string.action_ok)).assertIsEnabled()
+        composeTestRule.onNodeWithText(string(R.string.action_ok)).performClick()
+
+        assertThat(addedAmount).isEqualTo(5)
+    }
+
+    @Test
+    fun `add cards for today OK stays disabled for any input when nothing can be added`() {
+        // Mirrors the screenshot scenario: "Available New: 0" - typing 222 must not enable OK.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/settings/SettingsContentTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/SettingsViewModelTest.kt [hunk 3]
```diff
@@ -178,6 +191,50 @@ class SettingsViewModelTest {
...
+            // remaining quota = 20 (newPerDay) + 0 (extra) - 0 (shown) = 20
+            // availableToAddToday = 50 (unseen) - 20 (remaining quota) = 30
+            assertThat(viewModel.availableToAddToday.value).isEqualTo(30)
+        }
+
+    @Test
+    fun `loadAvailableNewCount reports zero availableToAddToday when unseen count is at or below current quota`() =
+        runTest(mainDispatcherRule.testDispatcher) {
+            // Regression test for the reported bug: when the deck has fewer (or no) unseen
+            // cards than the current quota already claims, there is no room to add more.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/SettingsViewModelTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/screens/settings/SettingsContentTest.kt [hunk 4]
```diff
@@ -188,6 +188,60 @@ class SettingsContentTest {
         composeTestRule.onNodeWithText(string(R.string.settings_add_cards_for_today_title)).assertIsDisplayed()
     }
 
+    @Test
+    fun `add cards for today OK is disabled once entered value exceeds what can still be added`() {
+        // Regression test for the reported bug: entering a number greater than the
+        // available capacity must not be accepted.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/settings/SettingsContentTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/screens/settings/components/NumberInputDialogTest.kt [hunk 5]
```diff
@@ -169,6 +169,68 @@ class NumberInputDialogTest {
...
+        okButton.performClick()
+
+        verify(exactly = 0) { onValueConfirm(any()) }
+    }
+
+    @Test
+    fun `OK stays disabled for any input when maximum is zero`() {
+        // Mirrors the "Available New: 0" dialog state from the reported bug: with no
+        // capacity left, no typed value should be accepted.
+        setContent(
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/settings/components/NumberInputDialogTest.kt:214]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/dojo/DojoViewModelTest.kt [hunk 6]
```diff
@@ -248,6 +253,92 @@ class DojoViewModelTest {
...
+
+            assertThat(viewModel.uiState.value.newQuotaRemaining).isEqualTo(17)
+        }
+
+    @Test
+    fun `newQuotaRemaining caps an oversized extraNewToday boost at the unseen total`() =
+        runTest(mainDispatcherRule.testDispatcher) {
+            // Simulates a boost that was granted before the write-side clamp existed
+            // (or any other way an oversized value ends up persisted): the display must
+            // still never exceed reality.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/dojo/DojoViewModelTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/ui/dojo/DojoViewModelTest.kt [hunk 7]
```diff
@@ -248,6 +253,92 @@ class DojoViewModelTest {
             assertThat(state.newQuotaRemaining).isEqualTo(0)
         }
 
+    @Test
+    fun `newQuotaRemaining is capped at the actual number of unseen cards left in the deck`() =
+        runTest(mainDispatcherRule.testDispatcher) {
+            // Regression test for the reported bug: the "N new" counter must never claim
+            // more new cards exist than are actually unseen in the deck, even if
+            // newPerDay + extraNewToday - newShown computes to a larger number.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/dojo/DojoViewModelTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/data/repository/AddCardsForTodayIntegrationTest.kt [hunk 8]
```diff
@@ -137,4 +137,51 @@ class AddCardsForTodayIntegrationTest {
...
+            reviewNextNewCard()
+            assertThat(repository.hasAvailableItems()).isFalse()
+        }
+
+    @Test
+    fun `add cards for today grants nothing when there are no unseen cards left`() =
+        runTest {
+            // Regression test: the dialog accepted input even when "Available New: 0" -
+            // adding cards must be a no-op once the deck has no unseen cards left.
+            dayCountersStore.setNewPerDay(2)
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/data/repository/AddCardsForTodayIntegrationTest.kt:171]: remove comment

File: app/src/test/java/com/procrastilearn/app/data/repository/AddCardsForTodayIntegrationTest.kt [hunk 9]
```diff
@@ -137,4 +137,51 @@ class AddCardsForTodayIntegrationTest {
             reviewNextNewCard()
             assertThat(repository.hasAvailableItems()).isFalse()
         }
+
+    @Test
+    fun `add cards for today cannot grant more than the actual unseen cards in the deck`() =
+        runTest {
+            // Regression test for the reported bug: the user could type any number into
+            // "Add Cards For Today" - even one far larger than the number of unseen cards
+            // that exist - and it would be accepted and rendered as the day's new quota.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/data/repository/AddCardsForTodayIntegrationTest.kt]: remove comment

File: app/src/test/java/com/procrastilearn/app/data/local/prefs/DayCountersStoreTest.kt [hunk 10]
```diff
@@ -83,19 +83,79 @@ class DayCountersStoreTest {
...
+
+            assertThat(store.read().first().extraNewToday).isEqualTo(0)
+        }
+
+    @Test
+    fun addExtraNewTodayRepeatedAddsStopAtCapacityInsteadOfAccumulatingPastIt() =
+        runTest {
+            // Regression for the "Add Cards For Today" bug: each add alone might look
+            // like it's within bounds, but repeated adds must not accumulate past the
+            // number of cards that actually exist.
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/data/local/prefs/DayCountersStoreTest.kt]: remove comment
